godocjson.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/doc"
+	"go/parser"
+	"go/token"
+	"os"
+)
+
+// Func represents a function declaration.
+type Func struct {
+	Doc               string `json:"doc"`
+	Name              string `json:"name"`
+	PackageName       string `json:"packageName"`
+	PackageImportPath string `json:"packageImportPath"`
+	// Decl              *ast.FuncDecl
+
+	// methods
+	// (for functions, these fields have the respective zero value)
+	Recv string `json:"recv"` // actual   receiver "T" or "*T"
+	Orig string `json:"orig"` // original receiver "T" or "*T"
+	// Level int    // embedding level; 0 means not embedded
+}
+
+// Package represents a package declaration.
+type Package struct {
+	Type       string             `json:"type"`
+	Doc        string             `json:"doc"`
+	Name       string             `json:"name"`
+	ImportPath string             `json:"importPath"`
+	Imports    []string           `json:"imports"`
+	Filenames  []string           `json:"filenames"`
+	Notes      map[string][]*Note `json:"notes"`
+	// DEPRECATED. For backward compatibility Bugs is still populated,
+	// but all new code should use Notes instead.
+	Bugs []string `json:"bugs"`
+
+	// declarations
+	Consts []*Value `json:"consts"`
```

Comment:
In general, it's best to avoid using arrays of pointers, or as values in maps (as above). Just creates more stuff on the heap.